daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -77,7 +77,8 @@ final class Engine {
     */
   def submit(
       cmds: Commands,
-      transactionSeed: Option[crypto.Hash] = None
+      participantId: ParticipantId,
+      submissionSeed: Option[crypto.Hash],
```

Comment:
Looks like the submission seed is only used once to derive the transaction seed. Instead of passing `participantId` and `submissionSeed`, can we instead pass the `transactionSeed` directly?

Reply:
The `submissionSeed` is used twice:
- It is used to compute the `transactionSeed` 
- It is stored in the transaction itself, so the receiver of the transaction can verify the `transactionSeed` was derived properly.

Reply:
Hm, where is it actually stored in the transaction? Sorry if I'm missing something obvious.

Reply:
See @remyhaemmerle-da's [comment](https://github.com/digital-asset/daml/pull/4533#discussion_r380156077).